proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -190,6 +191,9 @@ public static Builder newBuilder() {
   private static final com.google.protobuf.Descriptors.FieldDescriptor.Type ENUM_TYPE =
       com.google.protobuf.Descriptors.FieldDescriptor.Type.ENUM;
 
+  private static final ConcurrentMap<String, Map<Class<?>, Method>> mapOfMapOfMethods =
+      new MapMaker().makeMap();
```

Comment:
By the way, wouldn't Class -> Map generally allow for less garbage?  In the case where you serialize one class type, you end up with 1 outer map and 1 inner map of string -> method.  But with the String -> Map formulation, serializing one type would end you up with 1 outer map of String -> Map and N inner maps of String -> Class, one map per method.